scripts/prep-fle-addon.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+set -e
+set -x
+
+# This bash script expects FLE_NODE_SOURCE_PATH to be the path to a Node.js
+# checkout which contains OpenSSL, and where the relevant headers and library
+# files for the mongodb-client-encryption addon will be stored after
+# compiling them.
+
+# This isn't a lot, but hopefully after https://jira.mongodb.org/browse/WRITING-7164
+# we'll be able to simplify this further.
+
+BUILDROOT="$(dirname "$0")"/../tmp/fle-buildroot
+rm -rf "$BUILDROOT"
+mkdir -p "$BUILDROOT"
+cd "$BUILDROOT"
+BUILDROOT="$PWD"
+PREBUILT_OSNAME=''
+
+if [ x"$FLE_NODE_SOURCE_PATH" != x"" -a -z "$BUILD_FLE_FROM_SOURCE" ]; then
+  # Use prebuilt binaries for macOS and Windows. Linux builds are available,
+  # but may not match the correct OpenSSL version that we end up linking
+  # against.
+  case `uname` in
+      Darwin*)                          PREBUILT_OSNAME=macos;;
+      CYGWIN*|MINGW32*|MSYS*|MINGW*)    PREBUILT_OSNAME=windows-test;;
+  esac
+fi
+
+if [ x"$PREBUILT_OSNAME" != x"" ]; then
+  # Download and extract prebuilt binaries.
+  curl -L https://s3.amazonaws.com/mciuploads/libmongocrypt/all/master/latest/libmongocrypt-all.tar.gz | \
+    tar --strip-components=1 -xzvf - $PREBUILT_OSNAME
```

Comment:
This downloads prebuilt binaries from the latest version on the `master` branch of libmongocrypt, as I understand it, and the git commands below also pick up the latest `master`. Are we okay with that? I feel like if we pin specific versions of the dependencies, we’ll forget to update them, and adding some logic along the lines of “pick the most recent release version”, that’s going to come with some extra complexity that I’m not sure is worth it.

Reply:
I think that's fine, but I would double check with the team that manages it, so I think drivers?

Reply:
I asked this:

> Is the lack of versioning going to be a risk for us in any way? Should we pin to specific versions/tags?

and the answer was:

> You can just pin to a stable version of the c-driver to get the bson library. The flexibility of versions is for drivers that like the c-driver and those that derive from it.

Which I don’t feel tells us a lot, I feel? :smile:

Reply:
hmmm that answer was confusing, I think they possibly misunderstood your question

Reply:
Ok just read through the thread in #fle, I'm not sure I understand why specifically the bson library is relevant, since we'd care about pinning the entirety of libmongocrypt